File: src/main/java/ru/ezhov/rocket/action/types/AbstractRocketAction.java
package ru.ezhov.rocket.action.types;

import ru.ezhov.rocket.action.api.RocketActionConfiguration;
import ru.ezhov.rocket.action.api.RocketActionConfigurationProperty;
import ru.ezhov.rocket.action.api.RocketActionUi;

/**
 * Base class for UI action
 */
public abstract class AbstractRocketAction implements RocketActionUi, RocketActionConfiguration {

    protected RocketActionConfigurationProperty createRocketActionProperty(String name, String description, boolean required) {
        return new RocketActionConfigurationProperty() {

            @Override
            public String name() {
                return name;
            }

            @Override
            public String description() {
                return description;
            }

            @Override
            public boolean isRequired() {
                return required;
            }
        };
    }
}

File: src/main/kotlin/ru/ezhov/rocket/action/types/AbstractRocketAction.kt
package ru.ezhov.rocket.action.types

import ru.ezhov.rocket.action.api.RocketActionConfiguration
import ru.ezhov.rocket.action.api.RocketActionConfigurationProperty
import ru.ezhov.rocket.action.api.RocketActionFactoryUi

/**
 * Базовый класс для создания UI действия
 */
abstract class AbstractRocketAction : RocketActionFactoryUi, RocketActionConfiguration {
    protected fun createRocketActionProperty(
            key: String,
            name: String,
            description: String,
            required: Boolean
    ): RocketActionConfigurationProperty =
            object : RocketActionConfigurationProperty {
                override fun key(): String = key

                override fun name(): String = name

                override fun description(): String = description

                override val isRequired: Boolean = required
            }
}